### DISCUSSION TITLE
Allow MetaMask Mobile to switch networks?

### DISCUSSION THREAD

**Discussion by frolic** [wevm/wagmi#405] [Ideas]
`wallet_switchEthereumChain` does seem to work for MetaMask on iOS (though it doesn't send a push notification like Rainbow), but is disabled via https://github.com/tmm/wagmi/blob/5d87de6be33648036964394b371d93f709303ef9/packages/core/src/connectors/walletConnect.ts#L13

**Comment by tmm** [comment]
Correct! Added in https://github.com/tmm/wagmi/pull/406

**Reply by Jasmine-liang** [reply]
Hey tmm, juse wondering if there's any support for switching network in matamask(mobile) build-in browser. Not sure the pull request #406 is the right one. [Our product](https://thespace.game/claim) faces "Error switching network" on metamask mobile(Android).
Thanks in advance.

**Reply by tmm** [reply]
Yup, it should work! What version of wagmi are you using?

**Reply by Jasmine-liang** [reply]
It's ^0.4.12.

**Reply by tmm** [reply]
This is fixed in `wagmi@0.5`

**Reply by Jasmine-liang** [reply]
Hey, switching network works perfectly on metamask built-on browser. Does it also work on normal browser?
But I still get [Error switching network](https://62c698d6028b4f00083d9b06--radiant-frangipane-50d1a2.netlify.app/claim) on normal browser(FireFox, Chrome) on mobile(Android). 

Also, are your working on this issue:[wallet connect mobile linking keeps redirecting to the app store](https://github.com/WalletConnect/walletconnect-monorepo/issues/444#) ?
Just saw your reaction on the latest comment :)

**Reply by tmm** [reply]
Are you using MetaMask through WalletConnect on Firefox/Chrome? Also, what version of wagmi are you using?

**Reply by Jasmine-liang** [reply]
The first question: yes!
The second: the latest version `wagmi@0.5.9`

**Reply by Jasmine-liang** [reply]
So I'm wondering if it's supported through WalletConnect on Firefox/Chrome to connect to Metamask(mobile)

**Reply by tmm** [reply]
Just tried connecting to MetaMask for iOS with WalletConnect on Firefox and Chrome. I was able to switch chains programmatically with wagmi and through the MetaMask app.
